conn.go

```diff
@@ -1430,6 +1529,34 @@ func (c *Conn) executeQuery(ctx context.Context, qry *Query) *Iter {
 	case *resultVoidFrame:
 		return &Iter{framer: framer}
 	case *resultRowsFrame:
+		if x.meta.newMetadataID != nil {
+			// Updating the result metadata id in prepared stmt
+			//
+			// If a RESULT/Rows message reports
+			//      changed resultset metadata with the Metadata_changed flag, the reported new
+			//      resultset metadata must be used in subsequent executions
+
+			stmtCacheKey := c.session.stmtsLRU.keyFor(c.host.HostID(), c.currentKeyspace, qry.stmt)
+			oldInflight, ok := c.session.stmtsLRU.get(stmtCacheKey)
+			if !ok {
+				// We didn't find the stmt in the cache, so we just re-prepare it
+				return c.executeQuery(ctx, qry)
```

Comment:
Tbh this part is tricky for me because I can't properly test it with C*. I think you have a point. If **RESULT/ROWS** contains both `<new_metadata_id>` and retrieved rows then it makes sense to return the data instead of re-executing the query, even if the statement is not presented in the cache.

Reply:
Yeah @lukasz-antoniak was having issues trying to test this too, I'll see if I can reproduce this after I'm done reviewing the v5 frame format

Reply:
Thanks a lot! It will help much

Reply:
I'm unable to get C* to set the METADATA_CHANGED flag... I opened https://issues.apache.org/jira/browse/CASSANDRA-20028 to track this. Meanwhile we are unable to test this so it's a bit risky to merge it (if C* fixes this in a new version then existing gocql users who may have upgraded to get v5 support might start seeing errors).

Reply:
That's a shame... I hope anybody from ASF will help us with this one.

> Meanwhile we are unable to test this so it's a bit risky to merge it 

I agree with you, a test lacking implementation is risky, I have to do something with this...